oximeter/db/src/client.rs

```diff
@@ -2101,148 +2631,114 @@ mod tests {
-    ) -> Result<(), Error> {
+    async fn recall_field_value_u16_test(client: &Client) -> Result<(), Error> {
         let field = FieldValue::U16(1);
         let as_json = serde_json::Value::from(1_u16);
-        test_recall_field_value_impl(address, field, as_json).await?;
+        test_recall_field_value_impl(field, as_json, client).await?;
         Ok(())
     }
 
-    async fn recall_field_value_i16_test(
-        address: SocketAddr,
-    ) -> Result<(), Error> {
+    async fn recall_field_value_i16_test(client: &Client) -> Result<(), Error> {
         let field = FieldValue::I16(1);
         let as_json = serde_json::Value::from(1_i16);
-        test_recall_field_value_impl(address, field, as_json).await?;
+        test_recall_field_value_impl(field, as_json, client).await?;
         Ok(())
     }
 
-    async fn recall_field_value_u32_test(
-        address: SocketAddr,
-    ) -> Result<(), Error> {
+    async fn recall_field_value_u32_test(client: &Client) -> Result<(), Error> {
         let field = FieldValue::U32(1);
         let as_json = serde_json::Value::from(1_u32);
-        test_recall_field_value_impl(address, field, as_json).await?;
+        test_recall_field_value_impl(field, as_json, client).await?;
         Ok(())
     }
 
-    async fn recall_field_value_i32_test(
-        address: SocketAddr,
-    ) -> Result<(), Error> {
+    async fn recall_field_value_i32_test(client: &Client) -> Result<(), Error> {
         let field = FieldValue::I32(1);
         let as_json = serde_json::Value::from(1_i32);
-        test_recall_field_value_impl(address, field, as_json).await?;
+        test_recall_field_value_impl(field, as_json, client).await?;
         Ok(())
     }
 
-    async fn recall_field_value_u64_test(
-        address: SocketAddr,
-    ) -> Result<(), Error> {
+    async fn recall_field_value_u64_test(client: &Client) -> Result<(), Error> {
         let field = FieldValue::U64(1);
         let as_json = serde_json::Value::from(1_u64);
-        test_recall_field_value_impl(address, field, as_json).await?;
+        test_recall_field_value_impl(field, as_json, client).await?;
         Ok(())
     }
 
-    async fn recall_field_value_i64_test(
-        address: SocketAddr,
-    ) -> Result<(), Error> {
+    async fn recall_field_value_i64_test(client: &Client) -> Result<(), Error> {
         let field = FieldValue::I64(1);
         let as_json = serde_json::Value::from(1_i64);
-        test_recall_field_value_impl(address, field, as_json).await?;
+        test_recall_field_value_impl(field, as_json, client).await?;
         Ok(())
     }
 
     async fn recall_field_value_string_test(
-        address: SocketAddr,
+        client: &Client,
     ) -> Result<(), Error> {
         let field = FieldValue::String("foo".into());
         let as_json = serde_json::Value::from("foo");
-        test_recall_field_value_impl(address, field, as_json).await?;
-        Ok(())
-    }
-
-    async fn recall_field_value_ipv4addr_test(
```

Comment:
Did we lose this test?

Reply:
Ok, I think that's reasonable for this particular test, thanks.

In general, I think minimizing the test duration is not the right tradeoff. The absolute duration numbers are quite small (a few seconds or 10s of seconds), so a large relative change doesn't make for a substantially worse developer experience. In any case, by running all the test _implementations_ in a single Rust `#[test]` function, we're losing the ability to run exactly which tests we want. I think this actually _slows down_ iteration, since I can't just run one test to confirm new features or bug fixes. I need to run all of them and sift through the output to find that for the test I'm actually working on. I feel that robust, complete, and repeatable testing is more important than _fast_ tests as we implement multi-node ClickHouse.